### DISCUSSION TITLE
Feature Request: Convert "class" param to "className"

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#521] [Ideas]
Since IDOM uses python dicts for HTML attributes, there isn't any technical limitations to using `"class"` as an equivalent to react's `"className"`.

`className` is used within React to avoid the `class` keyword, but that isn't needed within IDOM.

Automatically converting `{ "class": ... }` to `{ "className": ... }` in the backend would clean up the syntax a bit.

### Current Behavior
If `{ "class": ... }` is passed in, a warning is provided that it's an invalid parameter.

### New Behavior
If `{ "class": ... }` is detected, pop it from the dict and add its values to `{ "className": ... }`

**Comment by rmorshea** [comment]
I don't think this is something I'm going to change since the interpretation of attributes is up to the client implementation. It happens that we're using React and that it expects `className` instead of `class`, but if we were to use some other JS framework like Vue or Svelte that wouldn't be the case.

**Reply by Archmonger** [reply]
Yeah I see your point.

I might end up monkey patching that on Conreq's end.
